File: BOJ/BOJ - 7568.py
"""
author : Kwak Daeun
github : https://github.com/daeunkwak
title : 덩치
description : 구현, 브루트포스 알고리즘
"""

N = int(input())
ppl = []
res = []

for _ in range (N) :
    ppl.append(int, list(input().split()))

for i in range (len(ppl)) :
    cnt = 0
    for j in range (len(ppl)) :
        if ppl[i][0] < ppl[j][0] :
            if ppl[i][1] < ppl[j][1] :
                cnt += 1
    res.append(cnt + 1)

for cnt in res :
    print(cnt, end = " ")

File: BOJ/BOJ - 7568.java
/**
 author : Kwak Daeun
 github : https://github.com/daeunkwak

 title : 덩치
 description : 구현, 브루트포스 알고리즘
 */

import java.io.BufferedReader;
import java.io.InputStreamReader;
import java.io.IOException;

public class BOJ_7568 {
    public static void main(String[] args) throws IOException{
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));

        int N = Integer.parseInt(br.readLine());

        int[][] arr = new int[N][2];

        String[] str;
        for(int i = 0; i < N; i++){
            // 키, 몸무게 입력받기
            str = br.readLine().split(" ");
            arr[i][0] = Integer.parseInt(str[0]);
            arr[i][1] = Integer.parseInt(str[1]);
        }

        StringBuilder sb = new StringBuilder();

        for(int i = 0; i < N; i++){
            int rank = 1;

            for(int j = 0; j < N; j++){
                if (i == j){
                    continue;
                }
                if (arr[i][0] < arr[j][0] && arr[i][1] < arr[j][1]) {
                    rank++;
                }
            }
            sb.append(rank).append(' ');
            // System.out.println(sb);
        }
        System.out.println(sb);
    }
}









